src/app/models/feedback.py

```diff
@@ -0,0 +1,120 @@
+# import logging
+
+from sqlalchemy import func, text
+from sqlalchemy.engine import Result
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import aliased
+from sqlalchemy.sql.expression import select
+
+from src.app.views.response.feedback import Feedback as ResponseFeedback
+from src.app.views.response.feedback import FeedbackCount
+from src.core.database.models.feedback import DataModelPredictionFeedback as dbFeedback
+from src.core.database.models.player import Player as dbPlayer
+
+
+class Feedback:
+    def __init__(self, session: AsyncSession) -> None:
+        self.session = session
+        # self.logger = logging.getLogger("Feedback")
+
+    async def get_feedback(self, player_names: list[str]):
+        async with self.session:
+            # print(f"Player names: {player_names}")
+            feedback_voter: dbFeedback = aliased(dbFeedback, name="feedback_voter")
+            feedback_subject: dbFeedback = aliased(dbFeedback, name="feedback_subject")
+            player_name: dbPlayer = aliased(dbPlayer, name="player_name")
+
+            query = select(
+                [
+                    feedback_voter.vote,
+                    feedback_voter.prediction,
+                    feedback_voter.confidence,
+                    player_name.name,
+                    feedback_voter.feedback_text,
+                    feedback_voter.proposed_label,
+                ]
+            )
+            query = query.select_from(player_name)
+            query = query.join(
+                feedback_subject, feedback_subject.subject_id == player_name.id
+            )
+            query = query.join(
+                feedback_voter, feedback_voter.voter_id == player_name.id
+            )
+            query = query.where(
+                text("player_name.name IN :names").bindparams(
+                    names=player_names
+                )  # prevent sql injection
+            )
```

Comment:
this should be the query
```sql
feedback_voter: dbPlayer = aliased(dbPlayer, name="feedback_voter")
feedback_subject: dbPlayer = aliased(dbPlayer, name="feedback_subject")

query: Select = select(
	[
		func.count(func.distinct(feedback_subject.id)).label("count"),
		feedback_subject.possible_ban,
		feedback_subject.confirmed_ban,
		feedback_subject.confirmed_player,
		dbFeedback.vote,
	]
)
query = query.select_from(dbFeedback)
query = query.join(
	feedback_voter, dbFeedback.voter_id == feedback_voter.id
)
query = query.join(
	feedback_subject, dbFeedback.subject_id == feedback_subject.id
)
query = query.where(feedback_voter.name.in_(player_names))
query = query.group_by(
		feedback_subject.possible_ban,
		feedback_subject.confirmed_ban,
		feedback_subject.confirmed_player,
		dbFeedback.vote,
)
```

Reply:
wait, there is a specific function for this.  https://github.com/Bot-detector/public-api/blob/4c1180aa78f47297df01280f7c21c719b9b79d35/src/app/models/feedback.py#L69-L120